sklearn/externals/_lobpcg.py

```diff
@@ -383,7 +383,7 @@ def lobpcg(A, X,
         gramYBY = np.dot(blockVectorY.T.conj(), blockVectorBY)
         try:
             # gramYBY is a Cholesky factor from now on...
-            gramYBY = cho_factor(gramYBY)
+            gramYBY = cho_factor(gramYBY, check_finite=False)
```

Comment:
I think this may be non finite because we are not checking `blockVectorX` on line 302